sqlglot/expressions.py

```diff
@@ -998,6 +994,10 @@ def with_(
 }
 
 
+class Table(Expression):
+    arg_types = {"this": True, "db": False, "catalog": False, **QUERY_MODIFIERS}
```

Comment:
Should this only be "joins"?

Reply:
That's a good point, I think it's safe to handle only joins here but let me double check.

Reply:
If I'm not mistaken, by following MySQL's syntax (2nd link, starting at the table_references rule) you can derive:

```
SELECT * FROM (LATERAL (SELECT * FROM tbl) as t)
```

On the other hand, I tried two different syntax checkers and this doesn't seem to be valid MySQL. Am I missing something?

Reply:
Yeah that doesn't look like a valid query.

I just meant should we only be allowing the "joins" arg, not _all_ the QUERY_MODIFIERS

Reply:
Yeah that seems correct. I think it's sufficient to use just `_parse_joins` instead of all the query modifier parsers too.

 I'll make a PR to fix this.